### PR TITLE
Fine Tuning, Cleanup and Reinstantiation of Full Security Whitelisting Functionality

### PR DESCRIPTION
**Pull request by mgerstner**
Turns out quite some fine tuning was still missing for the full integration into openSUSE:Factory. This is rather urgent, I would like to wrap this up this week.

### CODE REVIEW

File: rpmlint/checks/FileDigestCheck.py [hunk 1]
```diff
@@ -62,52 +65,45 @@ def _get_digest_configuration_group(self, pkgfile):
...
         while stat.S_ISLNK(pkgfile.mode):
             pkgfile = pkg.readlink(pkgfile)
             if not pkgfile:
-                return (False, None)
+                return None
+
+        return pkgfile
 
-        digest_hash = digest['hash']
+    def _calcDigest(self, pkgfile, algorithm):
```
Comment (marxin) [inline rpmlint/checks/FileDigestCheck.py:89]: Please use standard Python naming for identifiers: `_calc_digest`.

File: rpmlint/checks/FileDigestCheck.py [hunk 2]
```diff
@@ -62,52 +65,45 @@ def _get_digest_configuration_group(self, pkgfile):
...
             return (True, None)
 
+        pkgfile = self._resolveLinks(pkg, path)
+        if pkgfile is None:
+            return (False, None)
+
+        file_digest = self._calcDigest(pkgfile, algorithm)
+        return (file_digest == digest['hash'], file_digest)
+
+    def _resolveLinks(self, pkg, path):
```
Comment (marxin) [inline rpmlint/checks/FileDigestCheck.py:80]: Please use standard Python naming for identifiers: `_resolve_links`.

**Review by marxin — CHANGES_REQUESTED** [review]
Thank you for the changes, I've read the patch series. Apart from the naming conventions, the pull request is fine.

File: rpmlint/checks/FileDigestCheck.py (hunk 2 above)
Comment (mgerstner) [reply rpmlint/checks/FileDigestCheck.py]: Thanks for the review. I rebased and force-pushed, the function names in code and also the commit messages are adjusted now.

**Review by marxin — APPROVED** [review]
LGTM! Thanks.

**Comment by marxin** [comment]
And I've just made a submit request to Factory: https://build.opensuse.org/request/show/919077.

**Pull request merged**